src/cascadia/TerminalApp/App.base.h

```diff
@@ -24,12 +24,12 @@ namespace winrt::TerminalApp::implementation
 
     private:
         bool _contentLoaded{ false };
-        std::shared_ptr<XamlMetaDataProvider> _appProvider;
-        std::shared_ptr<XamlMetaDataProvider> AppProvider()
+        std::shared_ptr<winrt::TerminalApp::XamlMetaDataProvider> _appProvider;
+        std::shared_ptr<winrt::TerminalApp::XamlMetaDataProvider> AppProvider()

```

Comment:
my only worry with this one is that... we're never supposed to use shared_ptr for winrt things.

We may want to just make this...

```
winrt::TerminalApp::XamlMetaDataProvider _appProvider{ nullptr };
```

and then just instantiate one down on line 32.

Reply:
(with `_appProvider = winrt::TerminalApp::XamlMetaDataProvider{}`)

Reply:
we will need to change the signature of AppProvider(), and here's my suggestion: make it return `const winrt::blah&`, and change the `AppProvider()->` above to `AppProvider().`. Meh. I don't llove this class.

Reply:
Updated the code now.